### PR TITLE
Parry Nerf

### PR DESCRIPTION
**Pull request by Sparrowbird611**
Removes the ability to parry with most items, and makes it less skill dependent.

Why it's good for the game:

Parrying is honestly a bit of a meme. I get it, it's fun to parry attacks with paper, but it's not a great mechanic. It's punishment for doing everything right, and it's effects are both incredibly common, and incredibly powerful. For instance, there was a time a GAS attacked me. I was bored and wanted to leave any way, so I didn't try too hard to win, yet I managed to parry all but two of his attacks and got him pinned to the floor, as a Diona. Parrying is still possible with some items, but it's rarer in general now.

🆑

Nerf: Parrying is no longer based upon skill. Sorry guys, you can't parry with cloaks like batman any more.
Tweak: Removes parrying from most weapons. Generally, only swords and sword like weapons can parry.

/🆑

### CODE REVIEW

**Comment by EldritchSigma** [comment]
Dunno really about nerfing actualy parries with weapons, specially the esword. Sure silly to parry with a large sheet of titanium or a papercup or an autocompressor while unconcious.

**Comment by Roland410** [comment]
Removing parry chances from actual weapons like the stunbaton etc. isn't really something that I support either, sure, parrying with a pill bottle is stupid (seen it yesterday, lol), but things that make sense to parry with should remain (especially the cane >:) ).

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -500,7 +500,7 @@ var/list/global/slot_flags_enumeration = list(
 /obj/item/proc/handle_shield(mob/user, var/damage, atom/damage_source = null, mob/attacker = null, var/def_zone = null, var/attack_text = "the attack")
 	var/parry_chance = get_parry_chance(user)
 	if(attacker)
-		parry_chance = max(0, parry_chance - 10 * attacker.get_skill_difference(SKILL_COMBAT, user))
+		parry_chance = max(0, parry_chance - 0 * attacker.get_skill_difference(SKILL_COMBAT, user))
```
Comment (Higgin) [inline code/game/objects/items.dm]: If you change the multiplier to 0, you can get rid of if(attacker) as well - skill difference will have no impact on parry chance.

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -516,7 +516,7 @@ var/list/global/slot_flags_enumeration = list(
 	. = base_parry_chance
 	if(user)
 		if(base_parry_chance || user.skill_check(SKILL_COMBAT, SKILL_ADEPT))
-			. += 10 * (user.get_skill_value(SKILL_COMBAT) - SKILL_BASIC)
+			. += 0 * (user.get_skill_value(SKILL_COMBAT) - SKILL_BASIC)
```
Comment (Higgin) [inline code/game/objects/items.dm]: Same here - remove if(user) and the changed line if you just want to make skills have no impact on parrying.

Comment (Higgin) [inline code/game/objects/items.dm]:
```suggestion
			. += 0 * (user.get_skill_value(SKILL_COMBAT) - SKILL_BASIC) //skills have no effect on parrying when multiplier is 0
```

File: code/game/objects/items.dm (hunk 1 above)
Comment (Higgin) [inline code/game/objects/items.dm]:
```suggestion
		parry_chance = max(0, parry_chance - 0 * attacker.get_skill_difference(SKILL_COMBAT, user)) //skills have no effect on parrying when multiplier is 0
```

Comment (Higgin) [inline code/game/objects/items.dm]:
```suggestion
		parry_chance = max(0, parry_chance - 0 * attacker.get_skill_difference(SKILL_COMBAT, user)) //skills have no effect when multiplier is 0
```

File: code/game/objects/items.dm (hunk 2 above)
Comment (Higgin) [inline code/game/objects/items.dm]:
```suggestion
			. += 0 * (user.get_skill_value(SKILL_COMBAT) - SKILL_BASIC) //skills have no effect when multiplier is 0
```

**Comment by Sparrowbird611** [comment]
Sorry, I should have clarified one thing in this PR. E sword and broad sword still have their parry chance, as I felt those two were appropriate. I only removed parry chance from armblades, stun batons, and canes.

And that's a good idea, I'll do it Higgins.

**Comment by Sparrowbird611** [comment]
It's just me personally, but I feel parry should be a special thing on special items, not on the standard issue weapon of security.

As for the cane removal part, as much as it pained to me remove it, (I'm a cane user) I felt something acquirable by everyone for a single point shouldn't give that much of an advantage.

**Comment by Sparrowbird611** [comment]
I've reverted all changes to the e sword, and altered my nerf to the broadsword parry, which now has a 40% chance to parry instead of 30% (Original was 50%.)

**Comment by Glitterhoof** [comment]
what about weapons like the titanium machete or vibro-katana? will these also keep their previous parry chances as it makes alot of sense for these to have them aswell

**Comment by Sparrowbird611** [comment]
Vibro katana I believe either had no parry chance, or used the broad sword chance. It's completely untouched. Machetes don't have a parry chance any more.

**Comment by Glitterhoof** [comment]
Why wont machetes have a parry chance anymore?

**Comment by Sparrowbird611** [comment]
Machetes are carried around by every member of the exploration force, along with the fact that you can craft a machete. They're just as common, if not more common than stun batons. I'd rather only see the rarer items have a parry chance.

**Comment by Sparrowbird611** [comment]
Changelogs added.

**Comment by Jaraci** [comment]
I'm not a fan of this. Parrying is cool, and turning it into a super niche feature that only really benefits antagonists or whoever else can actually get e-swords is ungamer.

**Comment by Sparrowbird611** [comment]
You can technically still parry with other weapons. I THINK this includes every sword type, including the ceremoniall swords, but I haven't tested that.

Parrying is fun for all of one person. Yes, I enjoy our batman memes of cloak parrying, but frankly, it's old. Combat turns less into a fight and more into praying to RNGesus if some one tries to stack their parry chance, something I've done to great effect in the past.

Parrying will still be an incredibly powerful feature, and it's what will set swords apart from the other powerful melee options. The E sword will be unique due to the fact that it can be stored easily, but other swords will be able to compare to it in terms of parry chance.

**Comment by Jeff-Dome** [comment]
you can parry laser rounds.
why?

**Comment by Roland410** [comment]
Thought that was fixed, shouldn't happen unless you're using an esword.

**Comment by Sparrowbird611** [comment]
~~Double Energy Sword that parries all lasers when?~~

**Comment by EldritchSigma** [comment]
Batons of all sorts should be able to parry melee if you know how to use it, same for telebatons. Machetes aswell. Anything that is an actual weapon ya know.

**Comment by Sparrowbird611** [comment]
I disagree, since those are incredibly common items with their own ups and downs. I feel parrying should be primarily a sword thing, since they're rarer and often harder to conceal.

Also, I didn't even know we had telescopic batons here.

**Comment by Roland410** [comment]
Heads get telescopic batons in general and they are very rare, as they're basically an infinite stunbaton without the pain part, just the knockdown and stun.

**Comment by Sparrowbird611** [comment]
Ah, haven't played head here yet, so I wasn't aware.

**Comment by Sparrowbird611** [comment]
Technically unrelated to the PR, but I'm insanely surprised we have an energy machete amongst our energy based melee weapons. This server really loves it's machetes.

File: code/game/objects/items/weapons/cane.dm [hunk 3]
```diff
@@ -10,7 +10,7 @@
 	w_class = ITEM_SIZE_SMALL
 	matter = list(MATERIAL_ALUMINIUM = 50)
 	attack_verb = list("bludgeoned", "whacked", "disciplined", "thrashed")
-	base_parry_chance = 30
```
Comment (Andromeda-K22) [inline code/game/objects/items/weapons/cane.dm]: Readd parry chance.

File: code/game/objects/items/weapons/material/misc.dm [hunk 4]
```diff
@@ -70,7 +70,7 @@
 	w_class = ITEM_SIZE_NORMAL
 	slot_flags = SLOT_BELT
 	default_material = MATERIAL_TITANIUM
-	base_parry_chance = 50
```
Comment (Andromeda-K22) [inline code/game/objects/items/weapons/material/misc.dm]: Readd parry chance.

File: code/game/objects/items/weapons/material/misc.dm [hunk 5]
```diff
@@ -82,7 +82,7 @@
 	name = "fabricated machete"
 	desc = "A long, machine-stamped blade with a somewhat ungainly handle. Found in military surplus stores, malls, and horror movies since before interstellar travel."
 	default_material = MATERIAL_STEEL
-	base_parry_chance = 40
```
Comment (Andromeda-K22) [inline code/game/objects/items/weapons/material/misc.dm]: Readd parry chance.

File: code/game/objects/items/weapons/material/twohanded.dm [hunk 6]
```diff
@@ -27,7 +27,7 @@
 	var/base_icon
 	var/base_name
 	var/unwielded_force_divisor = 0.25
-	var/wielded_parry_bonus = 15
```
Comment (Andromeda-K22) [inline code/game/objects/items/weapons/material/twohanded.dm]: Readd parry chance.

File: code/game/objects/items/weapons/stunbaton.dm [hunk 7]
```diff
@@ -12,7 +12,7 @@
 	w_class = ITEM_SIZE_NORMAL
 	origin_tech = list(TECH_COMBAT = 2)
 	attack_verb = list("beaten")
-	base_parry_chance = 30
```
Comment (Andromeda-K22) [inline code/game/objects/items/weapons/stunbaton.dm]: Readd parry chance.

File: code/modules/augment/active/armblades.dm [hunk 8]
```diff
@@ -5,7 +5,7 @@
 	icon = 'icons/obj/augment.dmi'
 	applies_material_colour = 0
 	desc = "A handy utility blade for the discerning augmentee. Warranty void if used for cutting."
-	base_parry_chance = 30
```
Comment (Andromeda-K22) [inline code/modules/augment/active/armblades.dm]: Readd parry chance.

Comment (Sparrowbird611) [reply code/modules/augment/active/armblades.dm]: Kind of figred the advantage of an armblade was that it was undisarmable. However, given that it takes a fair amount of stuff to get one, I suppose that's fair.

Comment (Sparrowbird611) [reply code/modules/augment/active/armblades.dm]: I'll commit it later.

File: code/game/objects/items/weapons/material/twohanded.dm (hunk 6 above)
Comment (Sparrowbird611) [reply code/game/objects/items/weapons/material/twohanded.dm]: I frankly don't even recall removing this one. I think I lowered it though, but it still exists.

**Comment by Sparrowbird611** [comment]
The rest of those are exceptionally common weapons with high parry chances. They don't need the ability, they're all robust in their own right without the parry. They don't need a parry crutch.

**Comment by Sparrowbird611** [comment]
Oh snap, I forgot about this PR. I'll get back to it later today.

**Comment by Sparrowbird611** [comment]
Re added parry chances to all the weapons, but their numbers are lower. You can still parry with them, but it won't be as common.

**Comment by Sparrowbird611** [comment]
My baby is growing up so fast. He's a month old now. So precious.

**Comment by BurgerLUA** [comment]
For some reason this PR was left alone for a while, so it's getting a high priority tag.

**Comment by Sparrowbird611** [comment]
I technically didn't remove the parrying from weapons since andro requested otherwise, but I did lower it significantly.

**Comment by DerFlammenwerfer** [comment]
> Nerf: Parrying is no longer based upon skill.
I looked at the code for this, and it's just what it says on the tin. Your CQC skill will not provide any bonus to parry chance.

So the only real advantage to having a high CQC skill as a non-antag is being outright removed? Not only that, but weapons and objects which *could* conceivably parry in the hands of a capable wielder (steel rods, for instance) can't be used to parry either because you're restricting parry to just swords, **and not even sword-like bladed weapons?!**

The only reason this PR is polling so well is because you **LIED BY OMISSION** in the first sentence. You removed the bonus to parry that comes with CQC rather than, say, cutting it in half which would on its own probably do the job.

**Comment by Sparrowbird611** [comment]
I admit I could have worded that first sentence better, but to call it a lie is rather far, especially since I quite literally wrote in the change log "This removes parrying from skill."

You're underplaying the advantages of high CQC a bit. Grabs are still incredibly dangerous, and the higher your CQC, the easier you can escape them, or get others in them. I'd hardly say that's not something worth investing in.

I've pretty much argued why I restricted it mainly to swords all throughout this PR. Swords are, for the most part, rarer, harder to conceal, aside from the E sword. Parrying is their main advantage in this PR now, and I've given them all very high parry chances. I also gave it back to the armblade.

**Comment by Roland410** [comment]
Honestly, I still disagree with this PR, but the way you implemented is more of a crime than the concept. Not sure how I missed that.
Remove the code instead of setting them to 0 first of all.

**Comment by Sparrowbird611** [comment]
Alright, I'll do that.

**Comment by Sparrowbird611** [comment]
Something tells me there's a way to swap to different files and put those all into one commit, instead of literally making 4.

File: code/game/objects/items.dm [hunk 9]
```diff
@@ -500,7 +500,7 @@ var/list/global/slot_flags_enumeration = list(
 /obj/item/proc/handle_shield(mob/user, var/damage, atom/damage_source = null, mob/attacker = null, var/def_zone = null, var/attack_text = "the attack")
 	var/parry_chance = get_parry_chance(user)
 	if(attacker)
-		parry_chance = max(0, parry_chance - 10 * attacker.get_skill_difference(SKILL_COMBAT, user))
+		parry_chance = max(0, parry_chance - 0 * attacker.get_skill_difference(SKILL_COMBAT, user)) //skills have no effect when multiplier is 0
```
Comment (Roland410) [inline code/game/objects/items.dm:503]: Still applies here.

File: code/game/objects/items.dm [hunk 10]
```diff
@@ -516,7 +516,7 @@ var/list/global/slot_flags_enumeration = list(
 	. = base_parry_chance
 	if(user)
 		if(base_parry_chance || user.skill_check(SKILL_COMBAT, SKILL_ADEPT))
-			. += 10 * (user.get_skill_value(SKILL_COMBAT) - SKILL_BASIC)
+			. += 0 * (user.get_skill_value(SKILL_COMBAT) - SKILL_BASIC) //skills have no effect when multiplier is 0
```
Comment (Roland410) [inline code/game/objects/items.dm:519]: Here as well.

Comment (Sparrowbird611) [reply code/game/objects/items.dm]: Resolved

File: code/game/objects/items.dm (hunk 9 above)
Comment (Sparrowbird611) [reply code/game/objects/items.dm]: Resolved

File: code/game/objects/items.dm [hunk 11]
```diff
@@ -499,8 +499,6 @@ var/list/global/slot_flags_enumeration = list(
 //Otherwise should return 0 to indicate that the attack is not affected in any way.
 /obj/item/proc/handle_shield(mob/user, var/damage, atom/damage_source = null, mob/attacker = null, var/def_zone = null, var/attack_text = "the attack")
 	var/parry_chance = get_parry_chance(user)
-	if(attacker)
-	if(parry_chance)
 		if(default_parry_check(user, attacker, damage_source) && prob(parry_chance))
 			user.visible_message("<span class='danger'>\The [user] parries [attack_text] with \the [src]!</span>")
 			playsound(user.loc, 'sound/weapons/punchmiss.ogg', 50, 1)
```
Comment (Roland410) [inline code/game/objects/items.dm:504]: You definitely are, unless you wanted to make indentation errors.

Comment (Sparrowbird611) [reply code/game/objects/items.dm:504]: Was I not supposed to remove those two?

File: code/game/objects/items.dm [hunk 12]
```diff
@@ -499,8 +499,7 @@ var/list/global/slot_flags_enumeration = list(
 //Otherwise should return 0 to indicate that the attack is not affected in any way.
 /obj/item/proc/handle_shield(mob/user, var/damage, atom/damage_source = null, mob/attacker = null, var/def_zone = null, var/attack_text = "the attack")
 	var/parry_chance = get_parry_chance(user)
-	if(attacker)
-		parry_chance = max(0, parry_chance - 10 * attacker.get_skill_difference(SKILL_COMBAT, user))
+	if(attacker)
```
Comment (Roland410) [inline code/game/objects/items.dm:502]: Empty if again.

Comment (Sparrowbird611) [reply code/game/objects/items.dm]: Deleted if(attacker).

**Comment by BurgerLUA** [comment]
I'm going to do something fucking weird. I know that you guys might be repeating yourselves, but I need the following two things in the next 2 days.

1. A reason why this PR should be merged.
2. A reason why this PR shouldn't be merged.

Put your best and final argument forward based on the recent changes. If you can convince me that this PR should be merged, then I will merge it. If you can convince me that this PR shouldn't be merged, then I won't merge it.

**Comment by Roland410** [comment]
I'd honestly say not to merge this PR in its current state.
Parrying is kind of a niche thing to have, and it should either be skill based or weapon based (personally think it should be skill based), but this PR basically completely removes it from being a thing which I entirely disagree with.

**Comment by Sparrowbird611** [comment]
Alright, fair enough burger.

Parrying is, in my experience, exceptionally common in any melee fight that isn't a practice fight in the holodeck. It's quite literally a punishment for success, and given the sheer numbers of weapons that currently can parry, (Standard issue of all of exploration, the machete, standard issue of all of security, the stun baton, literally a 2 point cane in the loadout, ETC) it's not hard to stack that chance extremely high, especially when you throw in the parry boost that CQC gives. Given that parrying already have very few ways to be countered, I see that as an issue.

The goal of this PR was to keep parrying powerful, yet also make it less obnoxious, and something you could predict more easily. Hence why I made the sword class weapons capable of parrying. Swords are easily recognizable, and aren't standard issue for most roles. This gives swords a very respectable niche, making them one of the best melee options, keeping parrying powerful, but also less obnoxious, and means that the two most rabid groups in the game don't have roundstart access to weapons that are, among other things, powerful parry boosters, high damage/agony weapons, easily concealable, and given to every single member of their department.

I honestly directly disagree on making parrying skill based, and prefer it would be weapon based. I feel this PR should go through because it makes parry less ridiculous in terms of what you can parry with, and also means melee fights revolve less around who has the higher parry stat weapon and CQC skill, and more about who is a better robuster in close quarters, and who took the time and effort to acquire a weapon that would give him the huge advantages that parrying grants you. CQC is still a relevant skill, but no longer grants such a huge advantage in terms of parrying that any one below your tier is at an enormous disadvantage.

**Pull request merged**